projects/Amlogic/patches/ffmpeg/ffmpeg-999-aspect-ratio.patch

```diff
@@ -0,0 +1,62 @@
+From e5d21ff782977b0fe79a60796dc9d973d98e692c Mon Sep 17 00:00:00 2001
+From: Maxime Jourdan <mjourdan@baylibre.com>
+Date: Sun, 9 Sep 2018 17:22:35 +0200
+Subject: [PATCH] avcodec: v4l2_context: set frame SAR using VIDIOC_CROPCAP
+
+---
+ libavcodec/v4l2_context.c | 26 +++++++++++++++++++++++++-
+ 1 file changed, 25 insertions(+), 1 deletion(-)
+
+diff --git a/libavcodec/v4l2_context.c b/libavcodec/v4l2_context.c
+index efcb0426e4..8bff58ca5d 100644
+--- a/libavcodec/v4l2_context.c
++++ b/libavcodec/v4l2_context.c
+@@ -501,6 +501,24 @@ static int v4l2_get_coded_format(V4L2Context* ctx, uint32_t *p)
+     return 0;
+ }
+ 
++static AVRational v4l2_get_sar(V4L2Context* ctx)
++{
++    struct AVRational sar = { 1, 1 };
++    struct v4l2_cropcap cropcap;
++    int ret;
++
++    memset(&cropcap, 0, sizeof(cropcap));
++    cropcap.type = ctx->type;
++
++    ret = ioctl(ctx_to_m2mctx(ctx)->fd, VIDIOC_CROPCAP, &cropcap);
```

Comment:
@Elyotna I'd like to upstream this patch as well.

Is it necessary to invoke `VIDIOC_CROPCAP` per dequeued frame, or can we fetch the SAR and store it on the context?

Right now we use `pfd.revents & POLLPRI` to get notified of resolution changes.. would that work for cropping/SAR changes as well?